java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java

```diff
@@ -127,25 +147,37 @@ public ResponseEntity<Object> deleteEntity(
         ResponseParams responseParams = new ResponseParams();
         Response response = new Response(Response.API_ID.DELETE, "OK", responseParams);
         try {
+            checkEntityNameInDefinitionManager(entityName);
             String tag = "RegistryController.delete " + entityName;
             watch.start(tag);
-            Vertex deletedEntity = registryHelper.deleteEntity(entityId, userId);
+            JsonNode jsonNode = registryHelper.readEntity(userId, entityName, entityId, false, null, false);
+            if (jsonNode != null && jsonNode.has(entityName) && jsonNode.get(entityName).has(OSSystemFields._osSignedData.name())) {
+                String signedData = jsonNode.get(entityName).get(OSSystemFields._osSignedData.name()).asText();
+                registryHelper.revokeExistingCredentials(entityName, entityId, userId, signedData);
+            }
+            Vertex deletedEntity = registryHelper.deleteEntity(entityName, entityId, userId);
             if (deletedEntity != null && deletedEntity.keys().contains(OSSystemFields._osSignedData.name())) {
-                registryHelper.revokeExistingCredentials(entityName, entityId, userId, deletedEntity.value(OSSystemFields._osSignedData.name()));
+                String signedData = deletedEntity.property(OSSystemFields._osSignedData.name()).toString();
+                registryHelper.revokeExistingCredentials(entityName, entityId, userId, signedData);
```

Comment:
Why revoking credentials again? In above steps revoke credentials is already there